pvanalytics/metrics.py

```diff
@@ -65,3 +65,43 @@ def performance_ratio_nrel(poa_global, temp_air, wind_speed, pac, pdc0,
     performance_ratio = pac.sum() / pdc.sum()
 
     return performance_ratio
+
+
+def _calc_pathlength(signal, freq='d'):
+    # utility function to calculate the arc length of a time series.
+    # used when calculating the variability index.
+    dt = signal.index.to_series(keep_tz=True).diff().dt.total_seconds()/60
+    dy = signal.diff()
+    d = (dy**2 + dt**2)**0.5
+    if freq is not None:
+        return d.resample(freq).sum()
+    return d.sum()
+
+
+def variability_index(measured, clearsky, freq=None):
+    """
+    Calculate the variability index.
+
+    Parameters
+    ----------
+    measured : Series
+        Time series of measured GHI. [W/m2]
+    clearsky : Series
+        Time series of the expected clearsky GHI. [W/m2]
+    freq : pandas datetime offset, optional
+        Aggregation period (e.g. 'D' for daily).  If not specified,
+        a single value of VI for the entire time series will be returned.
```

Comment:
```suggestion
    freq : pandas datetime offset, optional
        Aggregation period (e.g. 'D' for daily).  If not specified,
        the variability index for the entire time series will be returned.
```

The default in `_calc_pathlength` is 'd', is 'd' as calendar day? I can't locate documentation for these strings.

Reply:
Ah I meant to take that out.  I'll push another commit to remove the default value for that parameter. 

These magic string codes are documented in the table here: https://pandas.pydata.org/docs/user_guide/timeseries.html#dateoffset-objects